src/lipalightninglib.udl

```diff
@@ -81,46 +119,60 @@ callback interface RemoteStorageCallback {
 enum InitializationError {
     "AsyncRuntime",
     "ChainMonitorWatchChannel",
-    "ChainSync",
+    "ChainSync", // Layer 1 synchronization
     "ChannelMonitorBackup",
-    "EsploraClient",
+    "EsploraClient", // Layer 1 handling
     "KeysManager",
     "Logic",
-    "PeerConnection",
+    "PeerConnection", // Layer 2 connections
     "PublicKey",
     "SecretGeneration",
 };
 
 [Error]
 enum RuntimeError {
-    "ChainSync",
+    "ChainSync", // Layer 1 synchronization
     "InvalidAddress",
     "InvalidPubKey",
     "Logic",
-    "PeerConnection",
+    "PeerConnection", // Layer 2 connections
 };
 
 [Error]
 enum LipaError {
+    // Invalid input.
+    // Consider fixing the input and retrying the request.
     "InvalidInput",
+    // Recoverable problem (e.g. network issue, problem with en external service).
+    // Consider retrying the request.
     "RuntimeError",
+    // Unrecoverable problem (e.g. internal invariant broken).
+    // Consider suggesting the user to report the issue to the developers.
     "PermanentFailure",
 };
 
 interface LightningNode {
+    // Initiate the Lightning node and let it run in a background thread
     [Throws=InitializationError]
     constructor([ByRef] Config config, RedundantStorageCallback redundant_storage_callback, LspCallback lsp_callback);
 
+    // Request some basic info about the local node
     NodeInfo get_node_info();
 
+    // Get information about the fee charged by the LSP for opening new channels
+    // This does NOT check for established channels, but only for established TCP connections
     [Throws=LipaError]
     LspFee query_lsp_fee();
 
+    // Verify whether there is an established Layer 2 connection between the local node and a given node
     boolean connected_to_node([ByRef] NodeAddress node);
 
+    // Create an invoice to receive a payment with
```

Comment:
Feels redundant to me, because the parameters state that very clearly already.
The only information added here is that msat = millisats, which imo the developers should be aware of at this point.

Another proposition to provide information that might be lacking:
```suggestion
    // Create an invoice to receive a payment with
    // amount_msat - the smallest amount of millisats required for the node to accept the incoming payment (sender will have to pay fees on top of that amount)
    // description - describes the invoice for both the payer and the payee
```

Unsure if this is correct:
 * unsure how we are going to deal with overpayments
 * also not entirely sure whether the description is intended for both the payer and the receiver)

Reply:
Building on top of your proposal:

```suggestion
    // Create an invoice to receive a payment with:
    //    - amount_msat - the smallest amount of millisats required for the node to accept the incoming payment (sender will have to pay fees on top of that amount)
    //    - description - a description to be embedded into the created invoice
```

> unsure how we are going to deal with overpayments

me too

> also not entirely sure whether the description is intended for both the payer and the receiver)

I suppose it's the description that can be inserted into Bolt11 invoices with the goal of stating the payment purpose